node/flatpak-node-generator.py

```diff
@@ -1785,11 +1860,21 @@ async def main() -> None:
 
     print('Reading packages from lockfiles...')
     packages: Set[Package] = set()
+    rcfile_node_headers: Set[NodeHeaders] = set()
```

Comment:
I don't believe a set is actually going to work correctly. We can create a set of Packages because that's a NamedTuple, which defines the proper `__hash__` and `__eq__` methods for us. But, `NodeHeaders` doesn't have that, so this ends up functioning as a slightly more expensive list.

If de-duplicating values here is important, you could probably convert `NodeHeaders` to a `NamedTuple`, like:

```python
class NodeHeaders(NamedTuple):
    target: str
    disturl: str
    runtime: str = 'node'
```

then just add `NodeHeaders.with_guessed_disturl` or similar as a custom constructor that guesses the `disturl` value.

Reply:
Makes sense, will convert NodeHeaders to a NamedTuple subclass.

Reply:
Done.